yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -353,7 +353,7 @@ private[yarn] class YarnAllocator(
 
     } else if (missing < 0) {
       val numToCancel = math.min(numPendingAllocate, -missing)
-      logInfo(s"Canceling requests for $numToCancel executor containers")
+      logInfo(s"Canceled requests for $numToCancel executor container(s)")
```

Comment:
No it already did. Follow the messages:

16/06/07 18:53:48 INFO yarn.YarnAllocator: Driver requested a total number of 619 executor(s).
16/06/07 18:53:48 INFO yarn.YarnAllocator: Canceling requests for 4 executor containers
16/06/07 18:53:48 INFO yarn.YarnAllocator: Driver requested a total number of 614 executor(s).
16/06/07 18:53:48 INFO yarn.YarnAllocator: Canceling requests for 5 executor containers

So: 619 - 5  = 614

The Cancel msg is coming after the executors are already canceled and we already logged the new requested a total number.